bin/setup

```diff
@@ -14,22 +14,36 @@ EOS
   exit 1
 end
 
+def skip_ui_update?
+  ENV["CI"]
+end
+
+def skip_database_setup?
+  ENV["SKIP_DATABASE_SETUP"] || ENV["CI"]
+end
+
+def skip_test_reset?
+  ENV["SKIP_TEST_RESET"]
+end
+
 Dir.chdir(ManageIQ::Environment::APP_ROOT) do
   ManageIQ::Environment.ensure_config_files
 
   puts '== Installing dependencies =='
   ManageIQ::Environment.install_bundler
   ManageIQ::Environment.bundle_update
 
-  ManageIQ::Environment.while_updating_ui do
-    unless ENV["SKIP_DATABASE_SETUP"]
-      ManageIQ::Environment.create_database
-      ManageIQ::Environment.migrate_database
-      ManageIQ::Environment.seed_database
-    end
+  ui_thread = ManageIQ::Environment.update_ui_thread unless skip_ui_update?
 
-    ManageIQ::Environment.setup_test_environment unless ENV["SKIP_TEST_RESET"]
+  unless skip_database_setup?
+    ManageIQ::Environment.create_database
```

Comment:
won't this break now cause on ci it won't create the database?

Reply:
It still sets up the test environment, https://travis-ci.com/github/ManageIQ/manageiq-cross_repo-tests/builds/155829057

Reply:
that's cross-repo... I meant in a normal travis run... what will create the DB if not this line?

Reply:
This does the same thing that core's .travis.yml does: https://github.com/ManageIQ/manageiq/blob/master/.travis.yml#L25

Reply:
Oh interesting...then technically we should have a before_script like the UI is doing.

Reply:
We still want to do this from `bin/setup` for normal development though right?

Reply:
yeah rake test:vmdb:setup is a travis specific thing because it also sets up parallelization (i had forgotten about that), so we would also keep it here in bin/setup for regular devs and skip in CI.

This section is good :+1: